internal/util/idlocker.go

```diff
@@ -14,9 +14,11 @@ limitations under the License.
 package util
 
 import (
+	"fmt"
 	"sync"
 
 	"k8s.io/apimachinery/pkg/util/sets"
+	klog "k8s.io/klog/v2"
```

Comment:
use the new log functions from `util` instead please

Reply:
This is an error log, not defined in util

Reply:
@Yuggupta27 it seems logging errors need to get added

Reply:
The error doesn't have any depth as const value, we are using the Errorf from klog directly.

Reply:
There is a `klog.ErrorDepth()` function, just like `klog.WarningDepth()`. These should get added to our `util/log.go` so that we can use a common logging interface (and use klog internally there).

Not a blocker for this PR, but definitely something we should have as an improvement.

Reply:
Yeah, just now noticed, Thanks

Reply:
Sure, more logging functions are needed. I'll create an issue for this one and address this at the rest of the code too.